src/http/request.cr

```diff
@@ -68,4 +89,21 @@ class HTTP::Request
   private def uri
     (@uri ||= URI.parse(@resource)).not_nil!
   end
+
+  private def ips_from(header)
+    if ips = headers[header]? || headers["Http-#{header.tr("_", "-")}"]?
```

Comment:
`HTTP::Headers` should have independent lookup by now, https://github.com/manastech/crystal/blob/master/spec/std/http/headers_spec.cr